rootfs/etc/nginx/lua/metric_emitter.lua

```diff
@@ -0,0 +1,50 @@
+local socket = ngx.socket.udp
+local cjson = require('cjson')
+local assert = assert
+
+local _M = {
+  queue = {}
+}
+
+local function flush_queue()
+  local s = assert(socket())
+  assert(s:setpeername("127.0.0.1", 8000))
+  for _, v in ipairs(_M.queue) do
+    assert(s:send(v))
+  end
+  _M.queue = {}
+  assert(s:close())
+end
+
+function _M.call()
+  -- Create JSON Metrics Payload  --
+  local jsonPayload = cjson.encode({
+    host                 = ngx.var.host                             or "-",
+    status               = ngx.var.status                           or "-",
+    remoteAddr           = ngx.var.realip_remote_addr               or "-",
+    remoteUser           = ngx.var.remote_user                      or "-",
+    bytesSent            = tonumber(ngx.var.bytes_sent)             or -1,
+    protocol             = ngx.var.server_protocol                  or "-",
+    method               = ngx.var.request_method                   or "-",
+    path                 = ngx.var.uri                              or "-",
+    requestLength        = tonumber(ngx.var.request_length)         or -1,
+    requestDuration      = tonumber(ngx.var.request_time)           or -1,
+    upstreamName         = ngx.var.upstream                         or "-",
+    upstreamIP           = ngx.var.upstream_addr                    or "-",
+    upstreamResponseTime = tonumber(ngx.var.upstream_response_time) or -1,
+    upstreamStatus       = ngx.var.upstream_status                  or "-",
+    namespace            = ngx.var.namespace                        or "-",
+    ingress              = ngx.var.ingress_name                     or "-",
+    service              = ngx.var.service_name                     or "-",
+  })
+
+  table.insert(_M.queue, jsonPayload)
+
+  local ok, err = ngx.timer.at(0, flush_queue)
```

Comment:
what's the behavior of ngx.timer.at` when nginx is receiving more than 1024 rps? (the default for pending timers)
https://github.com/openresty/lua-nginx-module#lua_max_pending_timers

Reply:
It will return nil, together with an error (from the documentation you linked)

Reply:
@aledbf we chatted about this, we'll be addressing it. Great catch!

We need to follow the way we did it in [our fork](https://github.com/Shopify/ingress/blob/master/rootfs/etc/nginx/lua/util/defer.lua) - where we essentially queue up request logs and pop it all off at a timer event that runs every second.

@fmejia97 will be addressing that today :)

Reply:
@aledbf Also worth mentioning - we use a nginx module to expose the active number of running timers; possibly worth merging this upstream to https://github.com/openresty/lua-nginx-module? Wdyt?

Reply:
> Wdyt?

Sure

Reply:
Just did :) @ElvinEfendi